server/tests/src/test/java/org/infinispan/server/resilience/GracefulShutdownRestartIT.java

```diff
@@ -60,4 +64,49 @@ public void testGracefulShutdownRestart() {
          assertEquals(String.format("v%03d", i), hotRod.get(String.format("k%03d", i)));
       }
    }
+
+   @Test
+   public void testCrashBeforeFinishedStarting() throws Exception {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.clustering().cacheMode(CacheMode.DIST_SYNC).persistence().addSingleFileStore().segmented(false);
+      RemoteCache<Object, Object> hotRod = SERVER.hotrod().withServerConfiguration(builder).create();
+
+      for (int i = 0; i < 100; i++) {
+         hotRod.put(String.format("k%03d", i), String.format("v%03d", i));
+      }
+
+      ContainerInfinispanServerDriver serverDriver = (ContainerInfinispanServerDriver) SERVER.getServerDriver();
+
+      // Locks is built-in and uses partition handling. We wait for it to be available on both nodes and then stops
+      // one of the nodes. It will move availability to DEGRADED.
+      CountdownLatchLoggingConsumer consumer = new CountdownLatchLoggingConsumer(1, ".*LOCKS]\\[Scope=.*]ISPN100010.*");
+
+      serverDriver.stop(1);
+      serverDriver.restart(1, consumer);
+      consumer.await(60, TimeUnit.SECONDS);
+
+      // Stop the CM before finishing start.
+      serverDriver.stop(1);
+
+      // Start again without problems.
+      serverDriver.restart(1);
+
+      // All data still available.
+      for (int i = 0; i < 100; i++) {
```

Comment:
Shouldn't we have a check to ensure that the LOCKS cache is once again available?

Reply:
The health check in the REST endpoint also includes the internal caches. I'll update to assert it is DEGRADED after the stop and HEALTHY after the restart.